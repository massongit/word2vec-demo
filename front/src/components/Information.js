import React from "react"
import {Table} from "reactstrap"
import {FormattedNumber, intlShape} from "react-intl"
import {propTypesSimilar} from "./App"

/**
 * 計算結果表示部
 */
class Information extends React.Component {
    static propTypes = {
        ...propTypesSimilar,
        intl: intlShape.isRequired
    }

    /**
     * ヘッダー部を描画する
     * @returns {node} ヘッダー部
     */
    renderHeader() {
        return (
            <tr>
                {
                    [
                        "header.candidate",
                        "header.word",
                        "header.cosine"
                    ].map((id, i) => ( // 見出しを上下左右中央揃えで表示
                        <th
                            key={i}
                            className="text-center"
                            style={{
                                "verticalAlign": "middle"
                            }}
                        >
                            {
                                this.props.intl.formatMessage({
                                    id
                                })
                            }
                        </th>
                    ))
                }
            </tr>
        )
    }

    /**
     * コンテンツ部を描画する
     * @returns {any[]} コンテンツ部
     */
    renderContents() {
        return this.props.similar.map((w, j) => (
            <tr
                key={j}
            >
                {
                    [
                        this.props.intl.formatNumber(j + 1),
                        w.word
                    ].map((data, i) => (
                        <td
                            className="text-center"
                            key={i}
                        >
                            {data}
                        </td>
                    ))
                }
                <td className="text-right">
                    <FormattedNumber
                        value={w.cosine}
                    />
                </td>
            </tr>
        ))
    }

    render() {
        return (
            <Table striped bordered hover size="sm" className="mb-0">
                <thead>
                    {this.renderHeader()}
                </thead>
                <tbody>
                    {this.renderContents()}
                </tbody>
            </Table>
        )
    }
}

export default Information
