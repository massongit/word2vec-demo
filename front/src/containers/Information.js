import Information from "../components/Information"
import {connect} from "react-redux"
import {injectIntl} from "react-intl"

const mapStateToProps = state => ({
    similar: state.showSimilarWords.words.similar
})

/**
 * 計算結果表示部のContainer
 */
export default injectIntl(connect(mapStateToProps)(Information))
