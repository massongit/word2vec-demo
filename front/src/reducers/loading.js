import * as types from "../actions/types"
import {handleActions} from "redux-actions"

/**
 * ローディングActionのReducer
 */
export default handleActions(
    {
        // ローディングActionのとき
        [types.LOADING]: (state, action) => {
            if (typeof action.payload.loading === "boolean") {
                return {
                    loading: action.payload.loading
                }
            } else {
                return state
            }
        }
    },
    {
        loading: false
    }
)
