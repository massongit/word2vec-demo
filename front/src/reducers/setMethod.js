import * as types from "../actions/types"
import {handleActions} from "redux-actions"

/**
 * 手法セットActionのReducer
 */
export default handleActions(
    {
        // 手法セットActionのとき
        [types.SET_METHOD]: (state, action) => {
            if (action.payload.method) {
                return {
                    prevMethod: state.method,
                    method: action.payload.method
                }
            } else {
                return state
            }
        }
    }, {}
)
