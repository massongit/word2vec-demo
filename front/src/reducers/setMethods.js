import * as types from "../actions/types"
import {handleActions} from "redux-actions"

/**
 * 手法一覧セットActionのReducer
 */
export default handleActions(
    {
        // 手法一覧セットActionのとき
        [types.SET_METHODS]: (state, action) => {
            if (0 < action.payload.methods.length) {
                return {
                    methods: action.payload.methods
                }
            } else {
                return state
            }
        }
    },
    {
        methods: []
    }
)
