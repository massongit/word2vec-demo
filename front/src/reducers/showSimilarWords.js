import * as types from "../actions/types"
import {handleActions} from "redux-actions"

/**
 * Word2Vecによる計算結果の表示ActionのReducer
 */
export default handleActions(
    {
        // Word2Vecによる計算結果の表示Actionのとき
        [types.SHOW_SIMILAR_WORDS]: (state, action) => {
            if (action.payload.words && action.payload.words.similar) {
                for (const s of action.payload.words.similar) {
                    if (!s || !s.word || !s.cosine) {
                        return state
                    }
                }

                if (action.payload.keywords && action.payload.words && !action.payload.words.similar.includes(undefined)) {
                    return {
                        keywords: action.payload.keywords,
                        words: action.payload.words
                    }
                }
            }

            return state
        }
    },
    {
        keywords: {
            positive: [],
            negative: []
        },
        words: {
            positive: [],
            negative: [],
            similar: []
        }
    }
)
