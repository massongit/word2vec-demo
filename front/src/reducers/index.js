import loading from "./loading"
import setMethod from "./setMethod"
import setMethods from "./setMethods"
import showSimilarWords from "./showSimilarWords"
import {combineReducers} from "redux"

/**
 * ルートReducer
 */
export default combineReducers({
    showSimilarWords,
    loading,
    setMethods,
    setMethod
})
