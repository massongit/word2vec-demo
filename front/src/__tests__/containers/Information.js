import React from "react"
import Information from "../../containers/Information"
import rootReducer from "../../reducers"
import {loadTranslation, mountWithIntl, shallowWithIntl} from "enzyme-react-intl"
import {createStore} from "redux"
import {showSimilarWordsState, word, word3} from "../../test_data"
import {dispatchActions, makeShowSimilarWordsAction} from "../reducers"
import words from "../../test_data/words"

/**
 * Storeを作成する
 * @returns {store} Store
 */
const makeStore = () => {
    const store = createStore(rootReducer)
    dispatchActions(store, makeShowSimilarWordsAction(showSimilarWordsState))
    return store
}

let informationComponent

loadTranslation("./src/translations/ja.json")

describe("containers/Information/snapshot", () => {
    it("初期状態からshowSimilarWordsへStateが遷移した際に、Componentが正しく配置されている", () => {
        informationComponent = shallowWithIntl(
            <Information
                store={makeStore()}
            />
        ).dive()
        expect(informationComponent).toMatchSnapshot()
    })
})

describe("containers/Information/other", () => {
    beforeEach(() => {
        informationComponent = mountWithIntl(
            <Information
                store={makeStore()}
            />
        )
    })

    it("初期状態からshowSimilarWordsへStateが遷移した際に、子要素にwordが含まれる", () => {
        expect(informationComponent.contains(word)).toBeTruthy()
    })

    it("初期状態からshowSimilarWordsへStateが遷移した際に、子要素にkeywordsが含まれる", () => {
        expect(informationComponent.contains(words.similar[0].word)).toBeTruthy()
    })
})
