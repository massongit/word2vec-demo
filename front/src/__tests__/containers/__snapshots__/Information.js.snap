// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Information/snapshot 初期状態からshowSimilarWordsへStateが遷移した際に、Componentが正しく配置されている 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Information)
    intl={
        Object {
            "defaultFormats": Object {},
            "defaultLocale": "en",
            "formatDate": [Function],
            "formatHTMLMessage": [Function],
            "formatMessage": [Function],
            "formatNumber": [Function],
            "formatPlural": [Function],
            "formatRelative": [Function],
            "formatTime": [Function],
            "formats": Object {},
            "formatters": Object {
              "getDateTimeFormat": [Function],
              "getMessageFormat": [Function],
              "getNumberFormat": [Function],
              "getPluralFormat": [Function],
              "getRelativeFormat": [Function],
            },
            "locale": "en",
            "messages": Object {
              "calculate": "計算",
              "description.input": "日本語の単語を空白で区切って入力し、計算ボタンをクリックしてください。",
              "description.input.supplement.1": "引き算を行う場合には、単語の先頭に『-』を付けます。",
              "description.input.supplement.2": "足し算を行う場合には、単語の先頭に『+』を付けることができます (基本的には付けなくても問題ありませんが、単語が『+』や『-』で始まる場合には必ず付けてください)。",
              "description.wordEmbedding": "単語をベクトル表現するための技術です。これを利用して、単語の足し算や引き算を行うことができます。",
              "errorMessage.default": "エラーが発生しました: {message}",
              "errorMessage.predict": "計算中にエラーが発生しました: {message}",
              "header.candidate": "候補",
              "header.cosine": "コサイン類似度 (-1 〜 1)",
              "header.keywords": "<キーワード>",
              "header.word": "単語",
              "title.description": "概要",
              "title.input": "入力",
              "title.result": "結果",
              "unknown": "(予測不能)",
            },
            "now": [Function],
            "textComponent": "span",
          }
    }
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "dispatch": [Function],
      "intl": Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {
          "calculate": "計算",
          "description.input": "日本語の単語を空白で区切って入力し、計算ボタンをクリックしてください。",
          "description.input.supplement.1": "引き算を行う場合には、単語の先頭に『-』を付けます。",
          "description.input.supplement.2": "足し算を行う場合には、単語の先頭に『+』を付けることができます (基本的には付けなくても問題ありませんが、単語が『+』や『-』で始まる場合には必ず付けてください)。",
          "description.wordEmbedding": "単語をベクトル表現するための技術です。これを利用して、単語の足し算や引き算を行うことができます。",
          "errorMessage.default": "エラーが発生しました: {message}",
          "errorMessage.predict": "計算中にエラーが発生しました: {message}",
          "header.candidate": "候補",
          "header.cosine": "コサイン類似度 (-1 〜 1)",
          "header.keywords": "<キーワード>",
          "header.word": "単語",
          "title.description": "概要",
          "title.input": "入力",
          "title.result": "結果",
          "unknown": "(予測不能)",
        },
        "now": [Function],
        "textComponent": "span",
      },
      "similar": Array [
        Object {
          "cosine": 0.6129051446914673,
          "word": "女王",
        },
        Object {
          "cosine": 0.5734513998031616,
          "word": "大王",
        },
        Object {
          "cosine": 0.5623782873153687,
          "word": "王女",
        },
        Object {
          "cosine": 0.557570219039917,
          "word": "王妃",
        },
        Object {
          "cosine": 0.536829948425293,
          "word": "デュエルモンスターズビギナーズパック",
        },
        Object {
          "cosine": 0.5333640575408936,
          "word": "デュエルモンスターズエキスパート",
        },
        Object {
          "cosine": 0.5321550369262695,
          "word": "戯",
        },
        Object {
          "cosine": 0.5298371911048889,
          "word": "遊戯",
        },
        Object {
          "cosine": 0.5250808000564575,
          "word": "デュエルモンスターズ",
        },
        Object {
          "cosine": 0.525033175945282,
          "word": "王家",
        },
      ],
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "dispatch": [Function],
        "intl": Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "formatDate": [Function],
          "formatHTMLMessage": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatPlural": [Function],
          "formatRelative": [Function],
          "formatTime": [Function],
          "formats": Object {},
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralFormat": [Function],
            "getRelativeFormat": [Function],
          },
          "locale": "en",
          "messages": Object {
            "calculate": "計算",
            "description.input": "日本語の単語を空白で区切って入力し、計算ボタンをクリックしてください。",
            "description.input.supplement.1": "引き算を行う場合には、単語の先頭に『-』を付けます。",
            "description.input.supplement.2": "足し算を行う場合には、単語の先頭に『+』を付けることができます (基本的には付けなくても問題ありませんが、単語が『+』や『-』で始まる場合には必ず付けてください)。",
            "description.wordEmbedding": "単語をベクトル表現するための技術です。これを利用して、単語の足し算や引き算を行うことができます。",
            "errorMessage.default": "エラーが発生しました: {message}",
            "errorMessage.predict": "計算中にエラーが発生しました: {message}",
            "header.candidate": "候補",
            "header.cosine": "コサイン類似度 (-1 〜 1)",
            "header.keywords": "<キーワード>",
            "header.word": "単語",
            "title.description": "概要",
            "title.input": "入力",
            "title.result": "結果",
            "unknown": "(予測不能)",
          },
          "now": [Function],
          "textComponent": "span",
        },
        "similar": Array [
          Object {
            "cosine": 0.6129051446914673,
            "word": "女王",
          },
          Object {
            "cosine": 0.5734513998031616,
            "word": "大王",
          },
          Object {
            "cosine": 0.5623782873153687,
            "word": "王女",
          },
          Object {
            "cosine": 0.557570219039917,
            "word": "王妃",
          },
          Object {
            "cosine": 0.536829948425293,
            "word": "デュエルモンスターズビギナーズパック",
          },
          Object {
            "cosine": 0.5333640575408936,
            "word": "デュエルモンスターズエキスパート",
          },
          Object {
            "cosine": 0.5321550369262695,
            "word": "戯",
          },
          Object {
            "cosine": 0.5298371911048889,
            "word": "遊戯",
          },
          Object {
            "cosine": 0.5250808000564575,
            "word": "デュエルモンスターズ",
          },
          Object {
            "cosine": 0.525033175945282,
            "word": "王家",
          },
        ],
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "intl": Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {
          "calculate": "計算",
          "description.input": "日本語の単語を空白で区切って入力し、計算ボタンをクリックしてください。",
          "description.input.supplement.1": "引き算を行う場合には、単語の先頭に『-』を付けます。",
          "description.input.supplement.2": "足し算を行う場合には、単語の先頭に『+』を付けることができます (基本的には付けなくても問題ありませんが、単語が『+』や『-』で始まる場合には必ず付けてください)。",
          "description.wordEmbedding": "単語をベクトル表現するための技術です。これを利用して、単語の足し算や引き算を行うことができます。",
          "errorMessage.default": "エラーが発生しました: {message}",
          "errorMessage.predict": "計算中にエラーが発生しました: {message}",
          "header.candidate": "候補",
          "header.cosine": "コサイン類似度 (-1 〜 1)",
          "header.keywords": "<キーワード>",
          "header.word": "単語",
          "title.description": "概要",
          "title.input": "入力",
          "title.result": "結果",
          "unknown": "(予測不能)",
        },
        "now": [Function],
        "textComponent": "span",
      },
    },
  },
}
`;
